app/src/main/java/de/danoeh/antennapod/fragment/SubscriptionFragment.java

```diff
@@ -192,6 +264,7 @@ private void loadSubscriptions() {
                 .observeOn(AndroidSchedulers.mainThread())
                 .subscribe(result -> {
                     navDrawerData = result;
+                    currentFeeds = (ArrayList<Feed>) navDrawerData.feeds;
```

Comment:
What is the relationship between `navDrawerData` and `currentFeeds`? Is `navDrawerData` still needed?

Why does this have to be an `ArrayList<Feed>`? I think it would be better to use `List<Feed>` everywhere instead of relying on the fact that `navDrawerData.feeds` currently returns an `ArrayList`. If `navDrawerData` is changed in the future, your code will break without a compiler error.